server/bundles/io.cloudbeaver.service.security/src/io/cloudbeaver/service/security/CBEmbeddedSecurityController.java

```diff
@@ -291,19 +291,76 @@ public SMUser getCurrentUser() throws DBException {
         return getUserById(getUserIdOrThrow());
     }
 
+    @Override
+    public int countUsers(String after, String userIdMask, Boolean enabledState) throws DBCException {
+        try (Connection dbCon = database.openConnection()) {
+            String whereString = "\nWHERE";
```

Comment:
filter construction logic seems overcomplicated and needs to be simplified

Reply:
moved to separate function